src/System Application/App/Vs Code Request Management/src/VSCodeRequestManagement.Codeunit.al

```diff
@@ -0,0 +1,147 @@
+﻿namespace System.Integration;
+
+using System.Apps;
+using System.Reflection;
+using System.Utilities;
+
+codeunit 8033 "VS Code Request Management"
+{
+    Access = Public;
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    var
+        AllObjWithCaption: Record AllObjWithCaption;
+        UriBuilder: Codeunit "Uri Builder";
+        VSCodeRequestHelper: DotNet VSCodeRequestHelper;
+        AlExtensionUriTxt: Label 'vscode://ms-dynamics-smb.al', Locked = true;
+        BaseApplicationIdTxt: Label '437dbf0e-84ff-417a-965d-ed2bb9650972', Locked = true;
+        SystemApplicationIdTxt: Label '63ca2fa4-4f03-4f2b-a480-172fef340d3f', Locked = true;
+        ApplicationIdTxt: Label 'c1335042-3002-4257-bf8a-75c898ccb1b8', Locked = true;
+
+    [Scope('OnPrem')]
+    procedure GetUrlToOpenExtensionSource(RepositoryUrl: Text[250]; CommitId: Text): Text
```

Comment:
this does not follow the  facade pattern we use in modules. Please make the Impl. internal and add a public facade with documentation